code/modules/projectiles/gun.dm

```diff
@@ -297,11 +299,13 @@
 		return FALSE
 
 	if(world.time < next_fire_time)
-		if (world.time % 3) //to prevent spam
+		if(world.time % 3 && !can_autofire) //to prevent spam
 			to_chat(user, SPAN_WARNING("\The [src] is not ready to fire again!"))
 		return FALSE
 
 	var/shoot_time = (burst - 1) * burst_delay
+	if(can_autofire)
+		shoot_time = DEFAULT_AUTOFIRE_BASE_DELAY * burst_delay
```

Comment:
That multiplication does not make sense.
Why would you multiply the DEFAULT_AUTOFIRE_BASE_DELAY (1) with the burst_delay.
A multiplication with 1 does not change the value in any way.